### PR TITLE
image uploader component and use it in edit profile component

### PR DESCRIPTION
**Pull request by SaifHayek**
related #65

### CODE REVIEW

File: client/src/components/Uploader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import Stack from '@mui/material/Stack'
+import './style.css'
+import { toast } from 'react-toastify'
+import { FormikProps } from 'formik'
+
+interface IUploaderProps {
+  name: string
+  formik:FormikProps<any>
+  btnName: string
+}
```
Comment (aaamra) [inline client/src/components/Uploader/index.tsx:14]: move to interfaces folder

File: client/src/components/Uploader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const Uploader:FC<IUploaderProps> = ({ name, formik, btnName }) => {
+  const [url, setUrl] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  const uploadImage = (e:any):void => {
+    const data = new FormData()
+    data.append('file', e.target.files[0])
+    data.append('upload_preset', 'shamskhodary')
+    data.append('cloud_name', 'dobsqpncd')
```
Comment (aaamra) [inline client/src/components/Uploader/index.tsx:23]: should be in config file

File: client/src/components/Uploader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    fetch('https://api.cloudinary.com/v1_1/dobsqpncd/image/upload', {
```
Comment (aaamra) [inline client/src/components/Uploader/index.tsx:25]: the link also should be in the config

File: client/src/components/Uploader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      method: 'post',
+      body: data,
+    })
+      .then((resp) => resp.json())
+      .then((data1) => {
+        setUrl(data1.url)
+        formik.setFieldValue(name, data1.url)
+        setLoading(false)
+      })
+      .catch(() => toast('Please, try again'))
```
Comment (aaamra) [inline client/src/components/Uploader/index.tsx:35]:
```suggestion
      .catch(() => toast.error('Please, try again'))
```

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by aaamra — APPROVED** [review]

**Review by MostafaBalousha123 — APPROVED** [review]

**Review by shamskhodary — APPROVED** [review]

**Review by Bakeza — APPROVED** [review]

**Pull request merged**